sympy/physics/mechanics/kane.py

```diff
@@ -366,10 +366,10 @@ def _initialize_kindiffeq_matrices(self, kdeqs, linear_solver='LU'):
             k_kqdot = kdeqs.jacobian(qdot)
             f_k = kdeqs.xreplace(u_zero).xreplace(qdot_zero)
 
-            # The kinematic differential equations should be linear in both q'
-            # and u, so check for u and q' in the components.
+            # The kinematic differential equations should be linear in u,
+            # so check for u in the components.
             dy_syms = find_dynamicsymbols(k_ku.row_join(k_kqdot).row_join(f_k))
-            nonlin_vars = [vari for vari in u[:] + qdot[:] if vari in dy_syms]
+            nonlin_vars = [vari for vari in u[:] if vari in dy_syms]
```

Comment:
The changes shouldn't be made to the kinematical differential equations, those do have both u's and qdot's.

Reply:
> The changes shouldn't be made to the kinematical differential equations, those do have both u's and qdot's.

oh, sh.....!!